docs(code): document filesystem tool allowlists (#5029)
langchain-ai/docs: src/oss/deepagents/code/cli-reference.mdx

The `-S`/`--shell-allow-list` flag applies in both interactive and non-interactive modes. Pass a comma-separated list of command names, `recommended` for safe read-only defaults, or `all` to permit any command. You can also set `DEEPAGENTS_CODE_SHELL_ALLOW_LIST` in the environment.

## Restrict filesystem tools

> By default, Deep Agents Code exposes all filesystem tools. Omit `--allow-fs-tools` or pass `all` to preserve that behavior. To expose only a subset, pass a comma-separated list:

Author: ```suggestion
By default, Deep Agents Code exposes all filesystem tools. To expose only a subset, pass a comma-separated list:
```

```bash
dcode -n "Audit this repository" --allow-fs-tools ls,read_file,glob,grep
```

> Valid names are `ls`, `read_file`, `write_file`, `edit_file`, `delete`, `glob`, `grep`, and `execute`. Explicit lists must include `read_file`. Empty lists, unknown names, and lists that combine `all` with named tools fail before the agent starts. The allowlist applies to the main agent and synchronous subagents in every session mode, but not to async subagents or non-filesystem tools.

Author: ```suggestion
Valid names are `ls`, `read_file`, `write_file`, `edit_file`, `delete`, `glob`, `grep`, and `execute`. Explicit lists must include `read_file`. The allowlist applies to the main agent and synchronous subagents in every session mode, but not to async subagents or non-filesystem tools.
```

The `-S`/`--shell-allow-list` flag applies in both interactive and non-interactive modes. Pass a comma-separated list of command names, `recommended` for safe read-only defaults, or `all` to permit any command. You can also set `DEEPAGENTS_CODE_SHELL_ALLOW_LIST` in the environment.

## Restrict filesystem tools

<Note>
    `--allow-fs-tools` requires `deepagents-code>=0.1.45`.

> </Note>

Author: ```suggestion
```

The `-S`/`--shell-allow-list` flag applies in both interactive and non-interactive modes. Pass a comma-separated list of command names, `recommended` for safe read-only defaults, or `all` to permit any command. You can also set `DEEPAGENTS_CODE_SHELL_ALLOW_LIST` in the environment.

## Restrict filesystem tools

By default, Deep Agents Code exposes all filesystem tools. To expose only a subset, pass a comma-separated list:

```bash
dcode -n "Audit this repository" --allow-fs-tools ls,read_file,glob,grep
```

Valid names are `ls`, `read_file`, `write_file`, `edit_file`, `delete`, `glob`, `grep`, and `execute`. Explicit lists must include `read_file`. The allowlist applies to the main agent and synchronous subagents in every session mode, but not to async subagents or non-filesystem tools.

<Note>
    `--allow-fs-tools` and `-S`/`--shell-allow-list` are two independent gates that both affect shell access:

    - **`--allow-fs-tools`** can decide whether the `execute` tool exists at all. Omit `execute` and the agent has no shell tool, regardless of `-S`.

>     - **`-S`/`--shell-allow-list`** decides which commands the `execute` tool is allowed to run. It has no effect when `execute` is not exposed.

Reviewer: I think this is slightly clearer. Why is this just about execute and not, for example also `glob`? Maybe we can explain that more clearly
```suggestion
    - **`--allow-fs-tools`** specifies whether the `execute` tool is available to the agent at all. For example, if you omit `execute` and the agent has no shell tool, regardless of `-S`/`--shell-allow-list`.
    - **`-S`/`--shell-allow-list`** specifies which commands the `execute` tool is allowed to run. It has no effect when `execute` is not exposed.
```
